extensions/replication/management.js

```diff
@@ -32,15 +43,23 @@ function putVersioning(bucketName, endpoint, cb) {
             Status: 'Enabled',
         },
     };
-    return getS3Client(endpoint).putBucketVersioning(params, err => {
-        if (err && err.code === 'NoSuchBucket') {
-            logger.info('cannot apply replication configuration: bucket ' +
-                        'does not exist',
-                        { sourceBucket: bucketName });
-            return cb();
+    
+    (async () => {
+        try {
+            const command = new PutBucketVersioningCommand(params);
+            await getS3Client(endpoint).send(command);
+            cb();
```

Comment:
you must NEVER invoke the continuation callback from within the `try` block !
this creates hard to debug issues when the "followup" code (in the callback) throws an exception....and thus invokes the callback as well!

either we "fully" migrate to async/await (probably too large a change to do along with SDK), or we need to stick with `promise.then()`... or use `utils.callbackify()`, which can make this migration even simpler as it does the combining of success and failures callbacks.